pkg/rancher-desktop/window/index.ts

```diff
@@ -167,6 +177,15 @@ export function openMain() {
     );
   }
 
+  window.on('closed', () => {
+    const cfg = loadSettings();
```

Comment:
Note that this is _subtly_ incorrect: `settings.load()` returns either the previously loaded settings, or loads them from disk (updating the in-memory copy in the process). However, `settings.save()` doesn't appear to update that in-memory copy.  This means calling `settings.load()` after a `settings.save()` in the same process will return the settings that were originally loaded from disk, missing any updates.

Please update `settings.save()` to update that in-memory copy, and add a comment saying that is required.

Reply:
Done. Good catch, that's a potentially problematic implementation detail that I wasn't aware of.